src/lib/pickles/pickles.ml

```diff
@@ -1257,6 +1354,153 @@ let%test_module "test no side-loaded" =
       assert (
         Promise.block_on_async_exn (fun () ->
             Tree_proof.Proof.verify_promise Tree_proof.example ) )
+
+    module Tree_proof_return = struct
```

Comment:
thanks for adding these examples. they make it very clear what this PR achieves!